### PR TITLE
Created ActionCell component

### LINKED ISSUE #4: Create Action Cell component
3 1 manage-users

### PR DESCRIPTION
**Pull request by andreytrofimov**
Closes #4

### CODE REVIEW

File: app/common/components/kendo-grid/editable-cell/get-action-cell.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import { GridCell, GridCellProps } from '@progress/kendo-react-grid';
+
+export type ActionsComponent = React.ComponentType<GridCellProps>;
+
+interface GetActionCellParams {
+    actions?: ActionsComponent;
+}
+
+export function getActionCell({ actions: Actions }: GetActionCellParams) {
```
Comment (xarety) [inline app/common/components/kendo-grid/editable-cell/get-action-cell.tsx]: This component doesn't look useful. Wouldn't it be better to use a closure with handlers here? 
```
function getActionCell(onEdit, onSave, onCancel, onDelete) {
    return class extends GridCell {
        // ...
    };
}
```

File: app/common/components/kendo-grid/kendo-grid-example.tsx [hunk 2]
```diff
@@ -175,7 +177,56 @@ export class GridExample extends React.Component {
         this.inEdit = false;
     }
 
+    @action
```
Comment (xarety) [inline app/common/components/kendo-grid/kendo-grid-example.tsx]: Redundant
```suggestion

```

File: app/common/components/kendo-grid/kendo-grid-example.tsx [hunk 3]
```diff
@@ -175,7 +177,56 @@ export class GridExample extends React.Component {
...
+    handleEditOneClick = (dataItem: Product) => {
+        this.gridState.edit(dataItem);
+    }
+
+    @action
```
Comment (xarety) [inline app/common/components/kendo-grid/kendo-grid-example.tsx]:
```suggestion

```

File: app/common/components/kendo-grid/kendo-grid-example.tsx [hunk 4]
```diff
@@ -175,7 +177,56 @@ export class GridExample extends React.Component {
...
+    handleSaveOneClick = (dataItem: Product) => {
+        this.gridState.saveEdit(dataItem);
+    }
+
+    @action
```
Comment (xarety) [inline app/common/components/kendo-grid/kendo-grid-example.tsx]:
```suggestion

```

File: app/common/components/kendo-grid/kendo-grid-example.tsx [hunk 5]
```diff
@@ -175,7 +177,56 @@ export class GridExample extends React.Component {
...
+    handleCancelOneClick = (dataItem: Product) => {
+        this.gridState.cancelEdit(dataItem);
+    }
+
+    @action
```
Comment (xarety) [inline app/common/components/kendo-grid/kendo-grid-example.tsx]:
```suggestion

```

File: app/common/components/kendo-grid/kendo-grid-example.tsx [hunk 6]
```diff
@@ -175,7 +177,56 @@ export class GridExample extends React.Component {
...
+    }
+
     render() {
+        const rowActions: ActionsComponent = ({ dataItem }) => {
+            const { inEdit } = dataItem;
+
+            if (inEdit) {
+                return (
+                    <React.Fragment>
+                        <Button small text onClick={this.handleCancelOneClick.bind(this, dataItem)}>
```
Comment (xarety) [inline app/common/components/kendo-grid/kendo-grid-example.tsx]: We avoid using bind, you could use a closure instead.

File: app/common/components/kendo-grid/editable-cell/get-action-cell.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    return class extends GridCell {
+        render() {
+            if (this.props.rowType !== "data") {
```
Comment (xarety) [inline app/common/components/kendo-grid/editable-cell/get-action-cell.tsx]: It won't work correctly with group footer rowType.

Comment (xarety) [reply app/common/components/kendo-grid/editable-cell/get-action-cell.tsx]: set `groupable={{ footer: 'visible' }}` for KendoGrid to look on issue.

**Review by xarety — APPROVED** [review]

File: app/common/components/kendo-grid/editable-cell/get-action-cell.tsx (hunk 7 above)
Comment (andreytrofimov) [reply app/common/components/kendo-grid/editable-cell/get-action-cell.tsx]: @xarety 
I fixed an issue for `groupFooter` ([74b5c5c](https://github.com/andreytrofimov/react-onboarding-practice-course/pull/7/commits/74b5c5c)), but there is another in `groupHeader`

image

Comment (andreytrofimov) [reply app/common/components/kendo-grid/editable-cell/get-action-cell.tsx]: Should we have a checkbox here?

Comment (xarety) [reply app/common/components/kendo-grid/editable-cell/get-action-cell.tsx]: Yup, this issue in the SelectColumnCell component, you can fix it in your branch if you want (same fix as you did for actions cell). It was fixed in the cactus release branch, a bit later we'll update the task template repository.

**Pull request merged**